File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <string.h>
#include <stdio.h>
#include <math.h>

bool is_armstrong_number(int candidate){
    char buffer[8] = {0};
    sprintf(buffer, "%d", candidate);
    int n = strlen(buffer);
    int sum = 0;
    for(int i=0; i<n; i++){
        sum += (int)pow((buffer[i] - '0'), n);        
    }
    return sum==candidate;
}

File: solutions/dart/armstrong-numbers/1/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(String number){
    var sum = BigInt.zero;
    final length = number.length;
    for(var i=0; i<length; i++){
      int di = int.parse(number[i]);
      sum += BigInt.from(di).pow(length);
    } 
    return BigInt.parse(number) == sum;
  }
}
